server/routes/veracode.ts

```diff
@@ -0,0 +1,87 @@
+import { type RequestHandler, Router } from 'express'
+import dayjs from 'dayjs'
+import asyncMiddleware from '../middleware/asyncMiddleware'
+import type { Services } from '../services'
+import { getFormattedName, veracodeFilters } from '../utils/utils'
+
+export default function routes({ serviceCatalogueService }: Services): Router {
+  const router = Router()
+
+  const get = (path: string, handler: RequestHandler) => router.get(path, asyncMiddleware(handler))
+  // const post = (path: string, handler: RequestHandler) => router.post(path, asyncMiddleware(handler))
+
+  get('/', async (req, res) => {
+    return res.render('pages/veracode')
+  })
+
+  get('/data', async (req, res) => {
+    const resultFilters = getResultFilters(req.query.results as string)
+    const exemptionFilters = getExemptionFilters(req.query.exemption as string)
+    const allComponents = await serviceCatalogueService.getComponents(exemptionFilters)
+    const passed = resultFilters.includes('passed')
+    const failed = resultFilters.includes('failed')
+    const unknown = resultFilters.includes('unknown')
+
+    const rows = allComponents
+      .filter(component => veracodeFilters(passed, failed, unknown, component.attributes.veracode_policy_rules_status))
+      .map(component => {
+        const hasVeracode = !!component.attributes.veracode_results_summary
+        const severityLevels = {
+          LOW: 0,
+          MEDIUM: 0,
+          HIGH: 0,
+          VERY_HIGH: 0,
+        }
+
+        component.attributes.veracode_results_summary?.severity?.forEach(severity => {
+          severity.category.forEach(category => {
+            // @ts-expect-error Suppress any declaration
+            severityLevels[category.severity] += category.count
+          })
+        })
+
+        let result = 'Failed'
+
+        if (!hasVeracode) {
+          result = 'N/A'
+        } else if (component.attributes.veracode_policy_rules_status === 'Pass') {
+          result = 'Passed'
+        }
+
+        return {
+          name: component.attributes.name,
+          hasVeracode,
+          result,
+          report: hasVeracode ? component.attributes.veracode_results_url : 'N/A',
+          date: hasVeracode
+            ? dayjs(component.attributes.veracode_last_completed_scan_date).format('YYYY-MM-DD HH:mm')
+            : 'N/A',
+          codeScore: hasVeracode ? component.attributes.veracode_results_summary['static-analysis'].score : 0,
+          severityLevels,
+        }
+      })
+
+    return rows
```

Comment:
Needs to be the same as the code you copied it from so `res.send(rows)` it's not a return 🙂

Reply:
Thanks Paul , corrected it now